sql/src/main/java/org/elasticsearch/action/bulk/BulkShardProcessor.java

```diff
@@ -202,6 +209,7 @@ private ShardId shardId(String indexName, String id, @Nullable String routing) {
             ).shardId();
         } catch (IndexMissingException e) {
             if (!autoCreateIndices) {
+                indicesDeleted.add(indexName);
```

Comment:
the exception is thrown in the next line which probably causes the whole process to be aborted anyway, why is it important to add the indexName to indicesDeleted here?